src/App.js

```diff
@@ -0,0 +1,109 @@
+import { Component } from 'react';
+import AppHeader from './AppHeader';
+import ItemsList from './ItemsList';
+import AppFooter from './AppFooter';
+
+class App extends Component {
+    constructor() {
+        super()
+        this.state = {
+            inputfield: '',
+            inputs: [],
+            saveButtonClassActive: false
+        };
+    }
+
+    componentDidMount() {
+        const storageItems = localStorage.getItem('items');
+        const parsedItems = JSON.parse(storageItems);
+        if (parsedItems) {
+            const items = [];
+            for (const item of parsedItems) {
+                const input = {
+                    value: item.value,
+                    isChecked: item.isChecked
+                };
+                items.push(input);
+            }
+            this.setState({ inputs: items });
+        }
+    }
+
+    onInputChange = event => {
+        this.setState({ inputfield: event.target.value })
+    }
+
+    onAddButtonClick = () => {
+        if (this.state.inputfield) {
+            this.setState(state => {
+                const item = {
+                    value: state.inputfield,
+                    isChecked: false
+                };
+                const inputs = state.inputs.concat(item);
+                return {
+                    inputs,
+                    inputfield: '',
+                };
+            }, () => {
+                if (this.state.saveButtonClassActive) {
+                    this.saveToLocalStorage();
+                }
+            })
+        } else {
+            alert("Fill the field");
+        }
+
+    }
+
+    onDeleteButtonClick = event => {
+        this.setState({
+            inputs: this.state.inputs.filter(function (item) {
+                if (item.value !== event.target.previousSibling.textContent) {
```

Comment:
please don't rely on `event.target.previousSibling.textContent` here.

The first reason for this, is that in general when you updating react state you should rely on the state itself or other logical objects that you introduce yourself and not DOM structure.

The second reason here would be that you create a tight implicit dependency between this function and where your delete button is placed. Imagine that you decide to add one more div between the delete button and the todo item. The change would look trivial however that change would lead to breaking this logic. In general that the real reason for the first reason 😊

Reply:
so, idea is that you should pass as parameter to this function not an event, but an item which user want to delete

Reply:
Ok. I used here as parameter the item.